### PR TITLE
*: remove SupportChunk() from interface ResultSet and RecordSet

### PR DESCRIPTION
**Pull request by XuHuaiyu**
As the title says.
PTAL
@zz-jason @coocood

### CODE REVIEW

File: tidb.go [hunk 1]
```diff
@@ -217,7 +217,7 @@ func GetRows4Test(ctx context.Context, sctx sessionctx.Context, rs ast.RecordSet
 		return nil, nil
 	}
 	var rows []types.Row
-	if sctx.GetSessionVars().EnableChunk && rs.SupportChunk() {
+	if sctx.GetSessionVars().EnableChunk {
```
Comment (ngaut) [inline tidb.go]: We should remove EnableChunk too.

Comment (zz-jason) [reply tidb.go]: will be done in next PRs.

Comment (XuHuaiyu) [reply tidb.go]: EnableChunk will be removed after removing Next.

**Pull request merged**